assignment4/IslandCounter.java

```diff
@@ -0,0 +1,60 @@
+package com.katermar.assignment4;
+
+/**
+ * Created by katermar on 6/21/2017.
+ */
+public class IslandCounter {
+    private char a[][];
+
+    public IslandCounter(char[][] a) {

```

Comment:
As per assignment input is a 2d array of booleans rather than chars. If you accept chars, you have to worry about someone putting 'x' or any other character there.